src/climate_learn/metrics/metrics.py

```diff
@@ -0,0 +1,407 @@
+# Standard Library
+from typing import Optional, Union
+
+# Local application
+from .utils import MetricsMetaInfo, register
+
+# Third party
+import numpy as np
+import torch
+import torch.nn as nn
+
+
+class Metric:
+    """Parent class for all ClimateLearn metrics."""
+    def __init__(
+        self,
+        aggregate_only: bool = False,
+        metainfo: Optional[MetricsMetaInfo] = None
+    ):
+        r"""
+        .. highlight:: python
+
+        :param aggregate_only: If false, returns both the aggregate and
+            per-channel metrics. Otherwise, returns only the aggregate metric.
+            Default if `False`.
+        :type aggregate_only: bool
+        :param metainfo: Optional meta-information used by some metrics.
+        :type metainfo: MetricsMetaInfo|None
+        """
+        self.aggregate_only = aggregate_only
+        self.metainfo = metainfo
+
+    def __call__(
+        self,
+        pred: torch.Tensor,
+        target: torch.Tensor
+    ) -> torch.Tensor:
+        """
+        :param pred: The predicted value(s).
+        :type pred: torch.Tensor
+        :param target: The ground truth target value(s).
+        :type target: torch.Tensor        
+
+        :return: A tensor. See child classes for specifics.
+        :rtype: torch.Tensor
+        """
+        raise NotImplementedError()
+
+
+class LatitudeWeightedMetric(Metric):
+    """Parent class for latitude-weighted metrics."""
+    def __init__(self, *args, **kwargs):
```

Comment:
Yes, it would throw an error. Writing `*args, **kwargs` was merely for my convenience. I can change this to explicitly say `aggregate_only` and `metainfo`.